### PR TITLE
matplotlib: Update to 3.9.2

### PR DESCRIPTION
**Pull request by infinitymdm**
**Summary**

Update matplotlib to version 3.9.2 and add monitoring.yml.

Resolves getsolus/packages#4343

There are a large number of enhancements in this update. See the [release notes](https://matplotlib.org/3.9.2/users/prev_whats_new/whats_new_3.9.0.html) for key details. The full changelog is available [here](https://github.com/matplotlib/matplotlib/compare/v3.8.2...v3.9.2).

**Test Plan**

Run the test code [here](https://matplotlib.org/stable/gallery/lines_bars_and_markers/simple_plot.html) to generate a simple plot

**Checklist**

- [x] Package was built and tested against unstable

### CODE REVIEW

**Comment by infinitymdm** [comment]
Briefly tested a couple reverse dependencies (`freecad`, `pitivi`, my own matplotlib-based code) and everything looks ok. The other packages that depend on matplotlib are far more specialized than I'll be able to effectively test.

Just to be sure though, I did a local (as in `go-task build-local`) rebuild of `graph-tool` and checked the `abi_used_*` files. The only changes were related to `libstdc++`, so I think this update should be fine?

**Comment by davidjharder** [comment]
I think this is fine, but let's wait until after sync to merge. This will give it more time in unstable

**Review by davidjharder — APPROVED** [review]

**Pull request merged**
